src/Microsoft.Health.Dicom.Core/Features/CustomTag/GetCustomTagsService.cs

```diff
@@ -0,0 +1,45 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Messages.CustomTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.CustomTag
+{
+    public class GetCustomTagsService : IGetCustomTagsService
+    {
+        private readonly ICustomTagStore _customTagStore;
+        private readonly IDicomTagParser _dicomTagParser;
+
+        public GetCustomTagsService(ICustomTagStore customTagStore, IDicomTagParser dicomTagParser)
+        {
+            EnsureArg.IsNotNull(customTagStore, nameof(customTagStore));
+            EnsureArg.IsNotNull(dicomTagParser, nameof(dicomTagParser));
+
+            _customTagStore = customTagStore;
+            _dicomTagParser = dicomTagParser;
+        }
+
+        public async Task<GetCustomTagResponse> GetCustomTagAsync(string tagPath, CancellationToken cancellationToken)
+        {
+            string internalTagPath = _dicomTagParser.ParseFormattedTagPath(tagPath);
```

Comment:
what if tagPath is Dicom Tag Keyword? like "DeviceSerialNumber"

Reply:
My understanding is that tagPath will always be the 8 digit tag id (multiple of 8 for lower levels). Has this expectation changed?

---
In reply to: [564890558](https://github.com/microsoft/dicom-server/pull/501#discussion_r564890558) [](ancestors = 564890558)